app/validation/validator.py

```diff
@@ -348,39 +356,97 @@ def _validate_skip_condition(self, skip_condition, answer_ids_with_group_id, blo
         errors.extend(self._validate_when_rule(when, answer_ids_with_group_id, block_or_group['id']))
         return errors
 
-    def _validate_list_collector_type(self, block):  # noqa: C901  pylint: disable=too-complex
+    def _validate_list_answer_references(self, block):
         errors = []
-        questions = []
-        add_answer_value = block['add_answer_value']
 
-        if 'question' in block:
-            questions.append(block['question'])
+        main_block_questions = self._get_all_questions_for_block(block)
+        main_block_ids = {answer['id'] for question in main_block_questions for answer in question['answers']}
+        remove_block_questions = self._get_all_questions_for_block(block['remove_block'])
+        remove_block_ids = {answer['id'] for question in remove_block_questions for answer in question['answers']}
 
-        if 'question_variants' in block:
-            for question in block['question_variants']:
-                questions.append(question)
+        if block['add_answer']['id'] not in main_block_ids:
+            errors.append(self._error_message(
+                'Add answer reference uses id not found in add_block: {}'.format(block["add_answer"]["id"]))
+            )
+        if block['remove_answer']['id'] not in remove_block_ids:
+            errors.append(self._error_message(
+                'Remove answer reference uses id not found in remove_block: {}'.format(block["remove_answer"]["id"]))
+            )
 
-        for question in questions:
-            for answer in question['answers']:
-                if answer['type'] != 'Radio':
+        return errors
+
+    def _validate_list_collector_type(self, block):  # noqa: C901  pylint: disable=too-complex, too-many-locals
```

Comment:
This is validating more than just the type, we could move the checking of `add_answer_value = block['add_answer']['value']` etc into the new `_validate_list_answer_references`.

Reply:
I've changed the name of the top level method to `_validate_list_collector` since the type doesn't have anything to do with it.

I'm reluctant to move the rest of the validation out since it relies on the data being built up in that function whereas validating the ids is fairly self contained.